testsuite/do_tests.sh.in

```diff
@@ -354,7 +354,11 @@ run_test ()
 # cmake during configuration.
 export PYTHONPATH="${NEST_PYTHONPATH}:${PYTHONPATH}"
 export PATH="${NEST_PATH}:${PATH}"
-export LD_LIBRARY_PATH="${NEST_LD_LIBRARY_PATH}:${LD_LIBRARY_PATH}"
+if test "x${INFO_OS}" = xDarwin ; then
+  export LYLD_LIBRARY_PATH="${NEST_LD_LIBRARY_PATH}:${LD_LIBRARY_PATH}"
```

Comment:
Letters got a bit garbled, and we need DY also on the right-hand side:

```suggestion
  export DYLD_LIBRARY_PATH="${NEST_LD_LIBRARY_PATH}:${DYLD_LIBRARY_PATH}"
```

Reply:
In the new nest_vars.sh, we use:
```sh
# The path where NEST shared libraries are generated. (Works also if $(DY)LD_LIBRARY_PATH is undefined.)
if [[ "$OSTYPE" == "darwin"* ]]; then
    export DYLD_LIBRARY_PATH="$NEST_INSTALL_DIR/@CMAKE_INSTALL_LIBDIR@/nest"${DYLD_LIBRARY_PATH:+:$DYLD_LIBRARY_PATH}
else
    export LD_LIBRARY_PATH="$NEST_INSTALL_DIR/@CMAKE_INSTALL_LIBDIR@/nest"${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
fi
```
1. Suggest to make appending to the path "safe" in case it is not defined using the above syntax
2. What is the proper test for being on MacOS? I copied the above from the Travis environment. (Possibly, they are equivalent.)

Reply:
> In the new nest_vars.sh, we use:
> 
> ```shell
> # The path where NEST shared libraries are generated. (Works also if $(DY)LD_LIBRARY_PATH is undefined.)
> if [[ "$OSTYPE" == "darwin"* ]]; then
>     export DYLD_LIBRARY_PATH="$NEST_INSTALL_DIR/@CMAKE_INSTALL_LIBDIR@/nest"${DYLD_LIBRARY_PATH:+:$DYLD_LIBRARY_PATH}
> else
>     export LD_LIBRARY_PATH="$NEST_INSTALL_DIR/@CMAKE_INSTALL_LIBDIR@/nest"${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
> fi
> ```
> 
>     1. Suggest to make appending to the path "safe" in case it is not defined using the above syntax
> 
>     2. What is the proper test for being on MacOS? I copied the above from the Travis environment. (Possibly, they are equivalent.)

@clinssen 
Technically, it is the same. We can also let CMake replace the variables. In do_test.sh.in. In addition, we need to add CMAKE_INSTALL_PREFIX. I will apply this to the code.